controllers/storageoscluster/metrics-exporter_operand.go

```diff
@@ -0,0 +1,153 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/kustomize/api/filesys"
+)
+
+const (
+	// metricsExporterPackage contains the resource manifests for metrics-exporter operand.
+	metricsExporterPackage = "metrics-exporter"
+)
+
+type MetricsExporterOperand struct {
+	name            string
+	client          client.Client
+	requires        []string
+	requeueStrategy operand.RequeueStrategy
+	fs              filesys.FileSystem
+	kubectlClient   kubectl.KubectlClient
+
+	prometheusCRDsInstalled bool
+	prometheusCRDsLock      chan struct{}
+}
+
+var _ operand.Operand = &MetricsExporterOperand{}
+
+func (c *MetricsExporterOperand) Name() string                                           { return c.name }
+func (c *MetricsExporterOperand) Requires() []string                                     { return c.requires }
+func (c *MetricsExporterOperand) RequeueStrategy() operand.RequeueStrategy               { return c.requeueStrategy }
+func (c *MetricsExporterOperand) PostReady(ctx context.Context, obj client.Object) error { return nil }
+
+// dependenciesFound dictates wether the dependencies are present
+// and thus we can apply the metrics-exporter resources
+// currently we depend on Prometheus CRDs
+func (c *MetricsExporterOperand) dependenciesFound() bool {
+	return c.prometheusCRDsInstalled
+}
+
+func (c *MetricsExporterOperand) updateDependenciesState(currentState bool) {
+	c.prometheusCRDsLock <- struct{}{}
+	defer func() {
+		<-c.prometheusCRDsLock
+	}()
+
+	c.prometheusCRDsInstalled = currentState
+}
+
+func (c *MetricsExporterOperand) ReadyCheck(ctx context.Context, obj client.Object) (bool, error) {
+	ctx, span, _, log := instrumentation.Start(ctx, "MetricsExporterOperand.ReadyCheck")
+	defer span.End()
+
+	ctx, cancel := context.WithTimeout(ctx, time.Minute)
+	defer cancel()
+
+	if !c.dependenciesFound() {
+		return false, nil
+	}
+
+	nodeDS := &appsv1.DaemonSet{}
+	key := client.ObjectKey{Name: "storageos-node", Namespace: obj.GetNamespace()}
+	if err := c.client.Get(ctx, key, nodeDS); err != nil {
+		return false, err
+	}
+
+	metricsDS := &appsv1.DaemonSet{}
+	key = client.ObjectKey{Name: "storageos-metrics-exporter", Namespace: obj.GetNamespace()}
+	if err := c.client.Get(ctx, key, metricsDS); err != nil {
+		return false, err
+	}
+
+	// metrics-exporter only returns complete metrics when there's one instance running
+	// alongside each node container. Failing to do so means the metrics can be missing
+	// data for some volumes.
+	if metricsDS.Status.NumberReady == nodeDS.Status.DesiredNumberScheduled {
+		log.Info("metrics-exporter is ready", "NumberReady", metricsDS.Status.NumberReady)
+		return true, nil
+	}
+
+	log.Info("metrics-exporter not ready")
+	return false, nil
+}
+
+func (c *MetricsExporterOperand) Ensure(ctx context.Context, obj client.Object, ownerRef metav1.OwnerReference) (eventv1.ReconcilerEvent, error) {
+	ctx, span, _, log := instrumentation.Start(ctx, "MetricsExporterOperand.Ensure")
```

Comment:
Does merics exporter a permanent all time component?

Reply:
Yes, as long as it can be installed

Reply:
Because i didn't find any configuration option to enable or disable :)

Reply:
We could something for clients that don't care about metrics but it's not in scope for now, let's do it at a later time

Reply:
I agree with Richard's point , we've found being able to disable features by default very helpful when doing releases.

I'm not aware of how difficult it is to implement, but if it's trivial I think we should do it. Perhaps @mhmxs can comment on the work required